Server/App/api/deck.py

```diff
@@ -256,10 +351,30 @@ def post(self, rating, user_id, connector, uuid):
         logging.info('{} rated deck {}'.format(user_id, uuid))
         return self.make_response(status=StatusCode.OK)
 
+class DeckShareHandler(BaseHandler):
+    @authorize_request_and_create_db_connector
+    @extract_user_id
+    def get(self, user_id, connector, uuid):
+        results = connector.call_procedure('FETCH_DECK', user_id, uuid)
+        if len(results) != 1:
+            return self.make_response(status=StatusCode.INTERNAL_SERVER_ERROR)
+        if results[0]['owner'] != user_id:
+            return self.make_response(status=StatusCode.FORBIDDEN)
+
+        while True:
```

Comment:
Have you tested this with a collision? You _might_ need a `try`/`except` around the `call_procedure`